src/main/java/xyz/nucleoid/codecs/MoreCodecs.java

```diff
@@ -225,15 +231,18 @@ private static <T> List<T> unitArrayList(T t) {
      * @param defaultSupplier a supplier for a default value if not present
      * @param <T>             the codec parse type
      * @return a {@link MapCodec} that decodes the specified field
+     * 
+     * @deprecated Use {@link Codec#optionalFieldOf(Codec, T)}
      */
+    @Deprecated
```

Comment:
All of the `propagatingOptionalFieldOf` methods I think we can deprecate - to be fair, there is a subtle difference in that these ones always encode default values. I don't think that was intentional, and certainly wasn't documented at least.